docs/hugo/content/design/ADR-2023-02-Helm-Chart-Size-Limitations.md

```diff
@@ -0,0 +1,108 @@
+---
+title: '2023-01: Helm Chart Size Limitations and Workarounds'
+---
+
+## Context
+
+Helm v3 stores chart state in a Kubernetes secret by default. Kubernetes secrets are limited to a max size based on 
+etcd's configuration. The default etcd configuration limits secrets to a maximum size of 1MB.
+
+Our chart secret is fast approaching 1MB. In fact, 
+[outstanding PRs](https://github.com/Azure/azure-service-operator/pull/2727) already cause us to surpass the 1MB limit.
```

Comment:
```suggestion
Our chart secret is fast approaching 1MB. In fact, outstanding PRs (including [#2727](https://github.com/Azure/azure-service-operator/pull/2727)] already cause us to surpass the 1MB limit.
```

Reply:
Are there any other PRs that should be listed as well?

Reply:
This is the only one that I know of that trips the limit currently, but others must be getting close.